lib/search_logic.rb

```diff
@@ -0,0 +1,133 @@
+module SearchLogic
+    class << self
+        def get_places_from_city_string(city_name, place_type)
+            begin
+                city = get_city_from_name(city_name)
+                places = city.public_send(place_type).to_a #city.attractions.to_a
+                return places if places.count == 20
```

Comment:
The `20` appears elsewhere in this module, and if you update one place without updating the other, then you could cause this module to always hit the API no matter what. I suggest putting this in a constant.

In case the `20` ever changes, it'd also be a good idea to put a limit so that if the number ever gets reduced while you already have places in your database, you don't display more places than you need, and you won't always hit Geoapify.

```suggestion
                places = city.public_send(place_type).limit(20).to_a #city.attractions.to_a
                return places if places.count >= 20
```